module-system/sov-state/src/containers/accessory_map.rs

```diff
@@ -57,8 +56,8 @@ where
     /// map’s key type.
     pub fn set<Q, S: Storage>(&self, key: &Q, value: &V, working_set: &mut AccessoryWorkingSet<S>)
     where
-        K: Borrow<Q>,
-        Q: Hash + Eq + ?Sized,
+        VC: EncodeLike<Q, K>,
```

Comment:
The type `VC` is a `StateValueCodec`, but here (and in other places) we're using it for keys instead. I see two problems with this:

1. `StateValueCodec` is only intended to (de)serialize values, and not keys, so it's a misnomer at this point.
2. Besides the misnomer nit, using the same (de)serialization methods for keys and values doesn't allow codec authors to use different encodings for keys and values, which is an important feature we want to preserve.

I see two main solutions:

- Two different traits, `StateKeyCodec` and `StateValueCodec` (similar to what we had before moving to `Hash` for keys).
- A single codec trait `StateCodec` with different methods for keys and values.

Reply:
Hmm, interesting. Do we have a motivating use case for separating the encodings of keys and values?

Reply:
Added!